src/base/static/state/ducks/ui.js

```diff
@@ -3,105 +3,102 @@
+  state.ui.isGeocodeAddressBarVisible;
+export const isEditModeToggled = state => state.ui.isEditModeToggled;
+export const uiVisibilitySelector = (uiComponentName, state) =>
+  state.ui.uiVisibility[uiComponentName];
+export const contentPanelComponentSelector = state =>
+  state.ui.contentPanelComponent;
+export const pageSlugSelector = state => state.ui.activePageSlug;
 
 // Actions:
-const SET_UI_RIGHT_SIDEBAR = "ui/SET_UI_RIGHT_SIDEBAR";
-const SET_UI_LEFT_SIDEBAR = "ui/SET_UI_LEFT_SIDEBAR";
-const SET_UI_CONTENT_PANEL = "ui/SET_UI_CONTENT_PANEL";
-const SET_CURRENT_TEMPLATE = "ui/SET_CURRENT_TEMPLATE";
+const UPDATE_CURRENT_TEMPLATE = "ui/UPDATE_CURRENT_TEMPLATE";
 const UPDATE_ADD_PLACE_BUTTON_VISIBILITY =
   "ui/UPDATE_ADD_PLACE_BUTTON_VISIBILITY";
-const UPDATE_MAP_CENTERPOINT_VISIBILITY =
-  "ui/UPDATE_MAP_CENTERPOINT_VISIBILITY";
 const UPDATE_GEOCODE_ADDRESS_BAR_VISIBILITY =
   "ui/UPDATE_GEOCODE_ADDRESS_BAR_VISIBILITY";
 const UPDATE_EDIT_MODE_TOGGLED = "ui/UPDATE_EDIT_MODE_TOGGLED";
+const UPDATE_UI_VISIBILITY = "ui/UPDATE_UI_VISIBILITY";
+const UPDATE_ACTIVE_PAGE = "ui/UPDATE_ACTIVE_PAGE";
+const UPDATE_CONTENT_PANEL_COMPONENT = "ui/UPDATE_CONTENT_PANEL_COMPONENT";
 
 // Action creators:
-export function setContentPanel(isOpen) {
-  return { type: SET_UI_CONTENT_PANEL, payload: isOpen };
-}
-export function setRightSidebar(isExpanded) {
-  return { type: SET_UI_RIGHT_SIDEBAR, payload: isExpanded };
+export function updateCurrentTemplate(templateName) {
+  return { type: UPDATE_CURRENT_TEMPLATE, payload: templateName };
 }
-export function setLeftSidebar(status) {
-  return { type: SET_UI_LEFT_SIDEBAR, payload: status };
-}
-export function setCurrentTemplate(templateName) {
-  return { type: SET_CURRENT_TEMPLATE, payload: templateName };
-}
-export function setAddPlaceButtonVisibility(isVisible) {
+export function updateAddPlaceButtonVisibility(isVisible) {
   return { type: UPDATE_ADD_PLACE_BUTTON_VISIBILITY, payload: isVisible };
 }
-export function setMapCenterpointVisibility(isVisible) {
-  return { type: UPDATE_MAP_CENTERPOINT_VISIBILITY, payload: isVisible };
-}
 export function setGeocodeAddressBarVisibility(isVisible) {
   return { type: UPDATE_GEOCODE_ADDRESS_BAR_VISIBILITY, payload: isVisible };
 }
 export function updateEditModeToggled(isToggled) {
   return { type: UPDATE_EDIT_MODE_TOGGLED, payload: isToggled };
 }
+export function updateUIVisibility(uiComponentName, isVisible) {
+  return {
+    type: UPDATE_UI_VISIBILITY,
+    payload: { uiComponentName, isVisible },
+  };
+}
+export function updateActivePage(pageSlug) {
+  return {
+    type: UPDATE_ACTIVE_PAGE,
+    payload: pageSlug,
+  };
+}
+export function updateContentPanelComponent(componentName) {
+  return {
+    type: UPDATE_CONTENT_PANEL_COMPONENT,
+    payload: componentName,
+  };
+}
 
 // Reducers:
 const INITIAL_STATE = {
-  isContentPanelOpen: false,
+  activePageSlug: null,
+  uiVisibility: {
```

Comment:
Here's how I'm planning to manage UI visibility. There is a general-purpose selector and action creator for interacting with these flags.